File: src/main/java/com/internship/tmontica_admin/banner/Banner.java
package com.internship.tmontica_admin.banner;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Banner {
    private int id;
    private String usePage;
    private boolean usable;
    private String imgUrl;
    private String link;
    private Date startDate;
    private Date endDate;
    private String creatorId;
    private int number;
}

File: front-end/src/types/banner.ts
import { IPagination } from "./common";

export interface IBanner {
  id: number;
  creatorId: string;
  link: string;
  number: number;
  startDate: string;
  endDate: string;
  usePage: string;
  imgUrl?: string | null;
  imgFile?: Blob | string;
  usable: boolean;
}

export interface IBannerPaging {
  banners: IBanner[];
  pagination: IPagination;
}
